shaders/clouds.omwfx

```diff
@@ -555,13 +560,14 @@ fragment main {
             float cloud_darken = pow(1.0 / (1.0 + fog.x), step);
             col.rgb =
                 // Attenuate light due to clouds
-                col.rgb * cloud_darken +
+                col.rgb +
                 // Re-add light that's been scattered
-                fog_light * fog.y * (1.0 - cloud_darken);
+                fog_light * fog.y * (2.0 - cloud_darken);
             */
 
             // Rayleigh scattering
-            col.rgb = mix(col.rgb, sky_color, (1.0 - 1.0 / (1.0 + step * 0.00001 * scattering_factor)) * clamp((15000 - spos.z) * 0.0001, 0.0, 1.0));
+            vec3 ambient = vec3(omw.ambientColor);
+            col.rgb = mix(col.rgb, (mix(mix(sky_color,(mix(mix(sky_color, mix(vec3(max(sun_attenuated.x, sky_color.x), max(sun_attenuated.y, sky_color.y), max(sun_attenuated.z, sky_color.z)), mix(max(ambient, sky_color), fog_color, normalize(pow((dot(sun_dir, dir) + 1), 2))), normalize(pow((dot(sun_dir, dir) + 1), 4))), (dot(dir, sun_dir) + 1)), sun_attenuated + sky_color, (sqrt(max(((dot(dir, sun_dir) + 1)/ 2), 1))) )), (dot(dir, sun_dir), + 1)), max(max(sky_color, ambient), fog_color), abs(sun_dir.z))), (1.0 - 1.0 / (1.0 + step * 0.00001 * scattering_factor * (2 + max((pow((dot(dir, sun_dir), + 1), 2)), 1)* 5))) * clamp((15000 - spos.z) * 0.0001, 0.0, 1.0));
```

Comment:
It's a bit hard to tell what's going on here, but it looks like a lot of it is invariant over the loop body and could be moved to before the loop, improving performance quite a bit (a lot of drivers are pretty bad at even basic optimisations like this, sadly). Any chance this could be done?